app/src/main/java/com/tomtom/sdk/examples/maps/ConfigurableMapActivity.kt

```diff
@@ -0,0 +1,198 @@
+     */
+    private fun initMap() {
+        val mapOptions = MapOptions(mapKey = apiKey)
+        mapFragment = MapFragment.newInstance(mapOptions)
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.map_container, mapFragment)
+            .commit()
+        mapFragment.getMapAsync { map ->
+            tomTomMap = map
+            mapView.contentDescription = applicationContext.resources.getString(R.string.map_ready)
+            enableUserLocation()
+        }
+    }
+
+    /**
+     * The SDK provides a LocationProvider interface that is used between different modules to get location updates.
+     * This examples uses the AndroidLocationProvider.
+     * Under the hood, the engine uses Android’s system location services.
+     */
+    private fun initLocationProvider() {
+        locationProvider = AndroidLocationProvider(context = this)
+    }
+
+    /**
+     * In order to show the user’s location, the application must use the device’s location services, which requires the appropriate permissions.
+     */
+    private fun enableUserLocation() {
+        if (areLocationPermissionsGranted()) {
+            showUserLocation()
+        } else {
+            requestLocationPermission()
+        }
+    }
+
+    /**
+     * The LocationProvider itself only reports location changes. It does not interact internally with the map or navigation.
+     * Therefore, to show the user’s location on the map you have to set the LocationProvider to the TomTomMap.
+     * You also have to manually enable the location indicator.
+     * It can be configured using the LocationMarkerOptions class.
+     *
+     * Read more about user location on the map in the Showing User Location guide.
+     */
+    private fun showUserLocation() {
+        initLocationProvider()
+        locationProvider.enable()
+        // zoom to current location at city level
+        onLocationUpdateListener = OnLocationUpdateListener { location ->
+            tomTomMap.moveCamera(CameraOptions(location.position, zoom = CAMERA_ZOOM_VALUE))
+            locationProvider.removeOnLocationUpdateListener(onLocationUpdateListener)
+        }
+        locationProvider.addOnLocationUpdateListener(onLocationUpdateListener)
+        tomTomMap.setLocationProvider(locationProvider)
+        val locationMarker = LocationMarkerOptions(type = LocationMarkerOptions.Type.Pointer)
+        tomTomMap.enableLocationMarker(locationMarker)
+        mapView.contentDescription = applicationContext.resources.getString(R.string.map_with_location)
+    }
+
+    private fun showDefaultLocation() {
+        val amsterdam = GeoPoint(AMSTERDAM_GEO_POINT_LATITUDE, AMSTERDAM_GEO_POINT_LONGITUDE)
+        val cameraOptions = CameraOptions(
+            position = amsterdam,
+            zoom = CAMERA_ZOOM_VALUE
+        )
+        tomTomMap.moveCamera(cameraOptions)
+        mapView.contentDescription = applicationContext.resources.getString(R.string.map_default_location)
```

Comment:
contentDescription is only for the accessiblity, do we really want to change it so often? From what I understand this app should be kept simple and easy to follow and not a perfect Android app